crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java

```diff
@@ -366,6 +380,14 @@ public void updatePredefinedFields(CRUDOperationContext ctx, JsonDoc doc) {
     }
 
     @Override
+    public Extension getExtensionInstance(Class<? extends Extension> extensionClass) {
+        if(extensionClass.equals(LockingSupport.class))
+            return new MongoLockingSupport(this);
+        else
+            return null;
```

Comment:
No, API spec says if it doesn't support it, it returns null.

Not all CRUD implementations will support all extensions.

On Thu, Jul 30, 2015 at 12:39 PM, Dennis Crissman <notifications@github.com>
wrote:

> In
> crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/135#discussion_r35904618>
> :
>
> > @@ -366,6 +380,14 @@ public void updatePredefinedFields(CRUDOperationContext ctx, JsonDoc doc) {
> >      }
> >
> >      @Override
> > +    public Extension getExtensionInstance(Class<? extends Extension> extensionClass) {
> > +        if(extensionClass.equals(LockingSupport.class))
> > +            return new MongoLockingSupport(this);
> > +        else
> > +            return null;
>
> Should it throw an exception, or maybe log a warning?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/135/files#r35904618>
> .
>